src/control-plane/backend/lambda/api/service/segment.ts

```diff
@@ -0,0 +1,337 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  CLICKSTREAM_SEGMENTS_CRON_JOB_RULE_PREFIX,
+  OUTPUT_USER_SEGMENTS_WORKFLOW_ARN_SUFFIX,
+  Segment,
+  SegmentDdbItem,
+  SegmentJobStatus,
+  SegmentJobTriggerType,
+} from '@aws/clickstream-base-lib';
+import {
+  DeleteRuleCommand,
+  EventBridgeClient,
+  PutRuleCommand,
+  PutTargetsCommand,
+  RemoveTargetsCommand,
+} from '@aws-sdk/client-eventbridge';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { SFNClient, StartExecutionCommand } from '@aws-sdk/client-sfn';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { NextFunction, Request, Response } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import { PipelineStackType } from '../common/model-ln';
+import { logger } from '../common/powertools';
+import { aws_sdk_client_common_config } from '../common/sdk-client-config-ln';
+import { ApiFail, ApiSuccess, BucketPrefix } from '../common/types';
+import { getBucketPrefix } from '../common/utils';
+import { CPipeline } from '../model/pipeline';
+import { DynamoDBSegmentStore } from '../store/dynamodb/dynamodb-segment-store';
+import { DynamoDbStore } from '../store/dynamodb/dynamodb-store';
+
+const segmentStore = new DynamoDBSegmentStore();
+const pipelineStore = new DynamoDbStore();
+const eventBridgeClient = new EventBridgeClient({
+  ...aws_sdk_client_common_config,
+  region: process.env.AWS_REGION,
+});
+const sfnClient = new SFNClient({
+  ...aws_sdk_client_common_config,
+  region: process.env.AWS_REGION,
+});
+const s3Client = new S3Client({
+  ...aws_sdk_client_common_config,
+});
+
+const API_FUNCTION_LAMBDA_ROLE = process.env.API_FUNCTION_LAMBDA_ROLE;
+export const SEGMENT_JOBS_LIST_LIMIT = 30;
+
+export class SegmentServ {
+  public async create(req: Request, res: Response, next: NextFunction) {
+    try {
+      const segment = this.constructSegmentFromInput(req, res);
+
+      // Create EventBridge rule
+      if (segment.refreshSchedule.cron !== 'Manual' && !!segment.refreshSchedule.cronExpression) {
+        const rule = await this.putEventBridgeRule(segment);
+        segment.eventBridgeRuleArn = rule.RuleArn;
+      }
+
+      // Save segment setting to DDB
+      const id = await segmentStore.create(segment);
+      logger.info('Create new segment setting, id: ' + id);
+
+      return res.status(201).json(new ApiSuccess({ id }, 'Segment created successfully.'));
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  public async list(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { appId } = req.query;
+      const segments = await segmentStore.list(appId as string);
```

Comment:
Do we support pagination?

Reply:
We don't support pagination for segment list yet. Can we assume that user won't have too many segments (like more than 100) for one single app?

Reply:
That's fine now.